yarn-project/pxe/src/pxe_service/pxe_service.ts

```diff
@@ -746,13 +706,17 @@ export class PXEService implements PXE {
     proofCreator: ProofCreator,
     msgSender?: AztecAddress,
   ): Promise<SimulatedTx> {
-    // Get values that allow us to reconstruct the block hash
-    const executionResult = await this.#simulate(txExecutionRequest, msgSender);
+    const simulateOperation = async () => {
+      // Get values that allow us to reconstruct the block hash
+      return new CacheableExecutionResult(await this.#simulate(txExecutionRequest, msgSender));
+    };
+    const {executionResult} = await withProverCache('PXEService.simulate', [txExecutionRequest], simulateOperation);
```

Comment:
Are we sure about caching this? `simulate` is not a pure action that depends just on the tx exec request, it depends also on the current state of the PXE. Feels like we should cache the proof, but not the simulation.

Reply:
That is fair, I guess it straightforwardly hides any changes to the simulator right now. Agreed, though it was somewhat useful as an aggressive mode perhaps. This would remove the need to cache txExecutionRequest

Reply:
Maybe an alternative would be to include the state of the simulator in the cache key? A combination of a checksum of the user's local db and the state root used for public simulation in the node (assuming public simulation is requested) could do the trick. But I think it's safer to not caching, since we should not be sending the same thing over the same state more than once after all.